src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -284,17 +283,20 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
           .build();
       tasks.add(task);
 
-      liveState.addStagingTask(task.getTaskId(), taskName);
-      persistentState.addHdfsNode(taskId, offer.getHostname(), taskType);
+      liveState.addStagingTask(task.getTaskId());
+      persistentState.addHdfsNode(taskId, offer.getHostname(), taskType, taskName);
     }
     driver.launchTasks(Arrays.asList(offer.getId()), tasks);
     return true;
   }
 
   private String getNextTaskType(String taskType) {
+    Collection<String> nameNodeTaskNames = persistentState.getNameNodeTaskNames().values();
```

Comment:
It would be better to put these inside the if statement so we're not contacting the zkState for both every time.